sys/include/net/eui_provider.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_eui_provider    IEEE EUI-48/64 provider
+ * @ingroup     net
+ * @brief       MAC address management
+ * @{
+ *
+ * @file
+ * @brief   EUI-48 and EUI-64 address provider
+ *
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+#ifndef NET_EUI_PROVIDER_H
+#define NET_EUI_PROVIDER_H
+
+#include "net/eui48.h"
+#include "net/eui64.h"
+#include "net/netdev.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Will match any device index
+ */
+#define NETDEV_INDEX_ANY    (0xFF)
+
+/**
+ * @brief   Function for providing a EUI-48 to a device
+ */
+typedef int (*netdev_get_eui48_cb_t)(const void *arg, eui48_t *addr);
+
+/**
+ * @brief   Function for providing a EUI-64 to a device
```

Comment:
It would be nice to mention that the function shall return 0 in case of success. Otherwise addr will be rejected.
The same for eui48 function.

Reply:
Added documentation for the function.